pilz_control/test/pjtc_test_helper.h

```diff
@@ -0,0 +1,177 @@
+static constexpr double STOP_TRAJECTORY_DURATION_SEC {0.2};
+static constexpr double GOAL_TIME_TOLERANCE_SEC {0.01};
+
+static constexpr double TIME_SIMULATION_START_SEC {0.1};
+static constexpr double DEFAULT_UPDATE_PERIOD_SEC {0.008};
+static constexpr unsigned int SLEEP_TIME_MSEC {5};
+static constexpr unsigned int WAIT_FOR_HOLD_FUTURE_MSEC {1000};
+
+using UpdateFunc = std::function<void()>;
+using GoalType = control_msgs::FollowJointTrajectoryGoal;
+
+template<typename T>
+bool isFutureReady(const std::future<T>& this_future)
+{
+  return this_future.wait_for(std::chrono::seconds(0)) == std::future_status::ready;
+}
+
+static void startSimTime(const ros::Time& start_time = ros::Time(TIME_SIMULATION_START_SEC))
+{
+  ros::Time::setNow(start_time);
+}
+
+//! @brief Make progress in simulated ros::Time.
+static void progressInTime(const ros::Duration& period)
+{
+  ros::Time current_time {ros::Time::now()};
+  ros::Time::setNow(current_time + period);
+}
+
+//! @brief Set parameters required for the controller tests
+static void setControllerParameters(const std::string& controller_ns)
+{
+  ros::NodeHandle controller_nh_ {controller_ns};
+
+  std::vector<std::string> joint_names(JOINT_NAMES.begin(), JOINT_NAMES.end());
+
+  controller_nh_.setParam(JOINTS_PARAMETER, joint_names);
+  controller_nh_.setParam(STOP_TRAJECTORY_DURATION_PARAMETER, STOP_TRAJECTORY_DURATION_SEC);
+  controller_nh_.setParam(GOAL_TIME_TOLERANCE_PARAMETER, GOAL_TIME_TOLERANCE_SEC);
+}
+
+static ros::Duration getGoalDuration(const control_msgs::FollowJointTrajectoryGoal& goal)
+{
+  return goal.trajectory.points.back().time_from_start;
+}
+
+/**
+ * @brief Cycle through three joint positions in order to always get a movement even if the previous goal failed.
+ */
+static GoalType generateSimpleGoal(const ros::Duration& goal_duration = ros::Duration(DEFAULT_GOAL_DURATION_SEC))
+{
+  static unsigned int position_index {0};
+  const std::vector<double> alternating_positions {0.5, -0.25, -0.5};
+  position_index = (++position_index) % alternating_positions.size();
+
+  std::vector<std::string> joint_names(JOINT_NAMES.begin(), JOINT_NAMES.end());
+
+  GoalType goal;
+  goal.trajectory.joint_names = joint_names;
+  goal.trajectory.points.resize(1);
+  goal.trajectory.points[0].time_from_start = goal_duration;
+  goal.trajectory.points[0].positions = {alternating_positions[position_index], 0.0};
+
+  return goal;
+}
+
+/**
+ * @brief Either return true, when the condition is fulfilled or false, when the timeout has been reached
+ * or ROS is shutdown.
+ * @param is_condition_fulfilled Boolean function, which is expected to return true eventually.
+ * @param timeout Timeout [ms] with respect to system time.
+ * @param update_func Update function. If non-empty, the following is done periodically:
+ * - Make progress in simulated ros::Time,
+ * - Invoke update_func.
+ */
+bool waitFor(const std::function<bool()>& is_condition_fulfilled,
```

Comment:
Is there a reason why all functions in this header are `static` except this one?

Reply:
No, I add will add it.